### ISSUE TITLE
fxFlex adds extra background animation to material buttons

### ISSUE THREAD

**Issue by levgaas** [angular/flex-layout#750]
#### Bug, feature request, or proposal:
Placing `fxFlex` adds background animation to buttons (material buttons)

#### What is the expected behavior?
No extra animations

#### What are the steps to reproduce?
[StackBlitz](https://stackblitz.com/edit/angular-8h1dcb)
To make it more visible, set animations speed to 10%

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular v6.0.3
Material v6.1.0
Chrome 66.0.3359.181

**Comment by CaerusKaru** [comment]
I removed the Flex Layout dependency and found the same behavior between both. I have no idea what you mean by set animations speed to 10%. Flex Layout does not have anything to do with animations, and does not directly interact with the Material library.

If you have a more specific reproduction that only works when Flex Layout is included, we can reopen this issue.

**Comment by levgaas** [comment]
@CaerusKaru I don't have such behavior without `flex-layout` dependency or `fxFlex` applied.

To make it more clear I recorded a video. 
https://drive.google.com/file/d/19EMQw4obqwfFCiOqiLFi4ftkqlA4REy6/view

One of the suggestions would be:
Material button indeed has transition for background. And color is dynamically set like
```
<button mat-raised-button
        color="primary">
  My Button
</button>
```

Like this [Like to Repo](https://github.com/angular/material2/blob/9bf720aa49898aa397ef1fb385faea2faa9618c2/src/lib/core/common-behaviors/color.ts#L33)

But somehow when `fxFlex` is applied it sets (applies) `mat-primary` css-class not from very beginning. Producing animation.

When I set `.mat-primary` class manually there is no initial animation.

I've found some issues with order of dynamic styles when `fxFlex` is applied. Could it be related?

Thanks
